lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -496,11 +496,77 @@ bool SessionState::apply_update_criteria(SessionStateUpdateCriteria& uc) {
   return true;
 }
 
+RuleStats SessionState::get_rule_delta(
```

Comment:
I think it's cleaner to attach a function like `get_delta_since_last_report` to the `RuleStats` struct.

Reply:
Its mostly map operations though not rulestats logic, not sure if I'm following